### ISSUE TITLE
Enhancement request: Ability to disable Lazy Loading

### ISSUE THREAD

**Issue by gmr** [grafana/grafana#17257]
With the v6.2 change to use lazy loading, I have users complaining about the amount of time it takes to scroll through large dashboards.

In looking through the documentation, I do not see any new configuration option to control the behavior.

**What would you like to be added**:

A setting that can be set either globally or at a per-dashboard level to toggle lazy-loading on/off.

**Why is this needed**:

The change in loading behavior has negatively impacted users without any way to revert to the previous loading behavior.

**Comment by szerwi** [comment]
I saw the same problem. Grafana loads only the part of the dashboard that is visible on the screen. When you scroll down, there is often need to wait a few seconds until the panels are loaded.

**Comment by CrashLaker** [comment]
Also need this. It would be ideal to have an option to disable lazy loading + sort of a graph render queue for it not to overload the backend server.

**Comment by dfredell** [comment]
Ditto. My use case is: I have a grafana dashboard in an iframe (yes I set the GF_SECURITY_ALLOW_EMBEDDING env var) of size 0 (css: width:0;height:0;border:0;) so it is invisible to the user but not the browser then I use jQuery to pull out values and display them on a ticker with a pretty background. Now after 6.2 only the first two single stat metrics are loaded. 

Having the ability to disable the lazy load for this grafana dashboard would be great.

Here is a snapshot of it after giving the iframe more size. 
![image](https://user-images.githubusercontent.com/8085744/60274544-d87fb600-98bd-11e9-9877-4517a6ec5620.png)

**Comment by CrashLaker** [comment]
@dfredell perhaps you could pull from each frame individually?

**Comment by dfredell** [comment]
@CrashLaker 1. If by 'frame' you mean 'single stat panel' that is how I'm doing it, I believe. The issue is the dashboard doesn't render panels that are 'out of view'.
Here is a test showing some issue (may or may not be the same as my pretty ticker issue):
I refresh my dashboard in the smallest Chrome window size.
![image](https://user-images.githubusercontent.com/8085744/60277864-049e3580-98c4-11e9-9da3-a1de26c2e259.png)
Then make the window normal size only 3 of the 7 single stat metrics are visible. Maybe this is a separate issue, or just some intended side effect?
![image](https://user-images.githubusercontent.com/8085744/60277952-2bf50280-98c4-11e9-8a55-15e55bc89640.png)
If I F5 refresh at the larger screen size all the single stat panels load like I expect.
![image](https://user-images.githubusercontent.com/8085744/60278085-62328200-98c4-11e9-9e86-671788925a97.png)




2. Or do you mean make a GET to the `api/datasources/proxy/2/api/v1/query_range?query=...` endpoint. That way would work but then I don't get the handy coloring and name translation that is setup on the Grafana panel. 

I found a workaround that works for me. I set the iframe to `width: 0; height:10000px;` that way all of the panels render but are still invisible to the user.

**Comment by CrashLaker** [comment]
@dfredell i mean sharing each frame as an "embeded iframe html".
![image](https://user-images.githubusercontent.com/4432491/60290297-d2ee9400-98ee-11e9-82a1-d9b4ce682bc3.png)

even if those views are dynamically rendered you can always grab those from your other website's frontend then iterate over them tweaking the url's parameters

**Comment by dfredell** [comment]
@CrashLaker Ahw I see now. Just the `panelId` changes in the url. That is a great way to do it, I will do that. 
Thanks

**Comment by ryantxu** [comment]
@dfredell I made a new issue for your report, see #17806

For lazy load options, I think there are three options that make sense:

1. off/disabled:  render and update everything all the time (grafana before 6.2)
2. lazy load: delay rendering until something is in view (grafana 6.2)
3. visible: (or something like that): don't make any updates unless the panel is visible.  In particular, dont execute a panel query if the panel is out of view

**Comment by tmm1** [comment]
An option to disable lazy loading for a dashboard would be very useful.

I'm using an embedded dashboard via iframe, and on iOS Safari I was having issues with two sets of scrollbars. I was able to fix this by injecting some css to disable the nested scrolling region in the iframe (`.scroll-canvas{ position: relative; }`), but that exposed a bug in the lazy load implementation which relies on scroll tracking to function.

**Comment by RedShift1** [comment]
If you don't mind compiling Grafana yourself, you can disable it as described here: https://davidsimpson.me/2020/06/10/disable-lazy-loading-charts-grafana/

**Comment by derIckeBrln** [comment]
Is it possible to include the hotfix mentioned by @RedShift1 into the source?
This still not makes it configurable per dashboard or panel, but at least allows a workaround until something like that is implemented in a clean way

**Comment by szibis** [comment]
The additional thing on my side is that when I run. Chrom on iOS is in most case not loading things when scrolling the screen down. It starts happening from around 8.x and disabling lazy loading in configuration could help me override this issue.

**Comment by kalinjul** [comment]
Same here, Grafana 8.2.5.
Widgets are missing in iOS Safari until i scroll all the way to the bottom and back to the top again (usually multiple times).
Same problem on my wall panel using chromium on a raspberry pi 7" display.
Didn't have the issue in 7.4.1.

**Comment by derIckeBrln** [comment]
The "solution" staed by @RedShift1 does not work this way anymore.
The files have been restructured.
Furthermore it has some drawbacks, because hidden Panels may and should still be lazyRendered for performance reasons:
![image](https://user-images.githubusercontent.com/13568207/146917761-47896154-8866-42d7-ab18-a532a4c39ea9.png)

So maybe the lazy-Rendering-Mechanism should imho be possible to be disabled (by dashboard):
- not at all
- completely
- only for not hidden panels

**Comment by Alex389-r** [comment]
I am having problems too with the lazy loading.

In my case I am calling an external API only once and putting all the data in a table panel. Then, I am displaying that data on different panels reusing the query from the table so I do not call my API more times that needed. 

In this case, because I am repeating my panels, when I scroll down, all the repeated panels keeps calling the API again and again.

¿Any solution?

**Comment by marcingminski** [comment]
Agree with @Alex389-r .  This is a big issue, especially on mobile devices, when users flip their fingers up to quickly scroll down to the bottom. Mobile devices will scroll the page much quicker than when using a mouse on a desktop. The lazy loading issues several queries to the source through the same shared panels and kills the back-end data store. The desktop version only does a single call per shared panel to the data store.

In this test dashboard, there are 4 "Source" panels that issue queries to the data store and several Stat panels that share data from the "Source" panels. On the desktop, as expected, we have 4 queries to the data store:
![image](https://github.com/grafana/grafana/assets/8288333/0db62958-6d74-4f63-b3df-997d884f80c8)

However, the mobile version kills the data store as the dashboard is scrolled down:
![image](https://github.com/grafana/grafana/assets/8288333/669eb95a-a1eb-4a13-9287-ea877432800b)

Summary. The lazy loading does not always improve performance and should be configurable.

**Comment by jonathansnofox** [comment]
This might be a different form of lazy loading. We have dashes with 100s of plots. We need a way to have the title appear for each of the panels so we can run control + f and find it. We'd prefer an option to keep lazy loading, but remove it from the plot titles and boundaries.

**Comment by GWhim** [comment]
I use an iframe to load the Grafana panel, and the main page and Grafana are cross domain. 
When I click the main page button, I expect to capture an image through a postmessage and display it. Nowadays, I find that there are problems with the images I capture. For example, when I adjust the screen resolution and some canvases are not visible on the screen, I will never be able to capture the complete image.
 All of this is due to the lazy loading mechanism, but I **cannot turn off** this mechanism.
